src/errors.rs

```diff
@@ -84,6 +84,17 @@ pub enum IllFormedError {
         /// Name of actually closed tag
         found: String,
     },
+    /// A comment contains forbidden double-hyphen (`--`) sequence inside.
+    ///
+    /// According to the [specification], for compatibility, comments MUST NOT contain
+    /// double-hyphen (`--`) sequence, in particular, they cannot end by `--->`.
```

Comment:
Is `--->` meant to be `-->`

Reply:
No, `--->`. This is due a fact that XML grammar too straightforward and disallow any `--` that does not immediately following by the `>`. Which means that when `--` is part of `--->`, it is disallowed